app/lib/kafka.server.ts

```diff
@@ -0,0 +1,34 @@
+/*!
+ * Copyright © 2023 United States Government as represented by the
+ * Administrator of the National Aeronautics and Space Administration.
+ * All Rights Reserved.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+import { Kafka } from 'gcn-kafka'
+
+import { getEnvOrDie } from './env.server'
+import type { Circular } from '~/routes/circulars/circulars.lib'
+
+const client_id = getEnvOrDie('KAFKA_CLIENT_ID')
+const client_secret = getEnvOrDie('KAFKA_CLIENT_SECRET')
+
+const producer = new Kafka({
+  client_id,
+  client_secret,
+  domain: 'dev.gcn.nasa.gov', // Intentionally set to dev for testing
+}).producer()
+
+export async function sendKafka(circular: Circular, topic: string) {
+  await producer.connect()
+  await producer.send({
+    topic,
+    messages: [
+      {
+        key: 'message',
+        value: JSON.stringify(circular),
+      },
+    ],
+  })
+  await producer.disconnect()
```

Comment:
Please close the Kafka connection only when the Lambda runtime is terminating. That way we can amortize the cost of opening the connection across multiple Lambda invocations, only paying it once per cold start.

Reply:
You probably _should_ call disconnect when the Lambda runtime is terminating gracefully. [For Python Lambdas, AWS sends a SIGTERM signal to the function.](https://docs.aws.amazon.com/lambda/latest/dg/lambda-python.html) The documentation doesn't say that it does that for Node.js Lamdbas, but it probably does.

Reply:
This might be a [useful example](https://github.com/aws-samples/graceful-shutdown-with-aws-lambda/blob/main/nodejs-demo/hello-world/app.js), I will see if I can implement something similar

Reply:
Actually I am not sure if this will work as we think. Without the disconnect, the function hangs and throws an error 
![image](https://github.com/nasa-gcn/gcn.nasa.gov/assets/11023698/a767e224-4fa9-4ce0-a3d4-1f8dda4d492f)

I'll do some more testing though to be sure

Reply:
Okay, I have something in the latest commit that should work. The listener gets added in, but locally there is no SIGTERM in sandbox. I'm not 100% confident in this implementation, but it should work